src/main/java/org/qualityannotate/core/MainCommand.java

```diff
@@ -1,115 +1,93 @@
 package org.qualityannotate.core;
 
+import io.quarkus.logging.Log;
 import jakarta.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
 import lombok.RequiredArgsConstructor;
 import org.qualityannotate.api.coderepository.CodeRepository;
 import org.qualityannotate.api.coderepository.Comment;
 import org.qualityannotate.api.coderepository.FileComment;
-import org.qualityannotate.api.qualitytool.GlobalMetrics;
-import org.qualityannotate.api.qualitytool.Issue;
 import org.qualityannotate.api.qualitytool.MetricsAndIssues;
 import org.qualityannotate.api.qualitytool.QualityTool;
-import org.qualityannotate.coderepo.github.GithubCodeRepository;
-import org.qualityannotate.quality.sonarqube.SonarqubeQualityTool;
+import org.qualityannotate.coderepo.CodeRepoProvider;
+import org.qualityannotate.coderepo.github.GithubConfig;
+import org.qualityannotate.quality.QualityToolProvider;
+import org.qualityannotate.quality.sonarqube.SonarqubeConfig;
 import picocli.AutoComplete;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Parameters;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Collectors;
-
-@Command(name = "qualityannotate", mixinStandardHelpOptions = true, subcommands = AutoComplete.GenerateCompletion.class,
-         description = """
-                 Helps extracting issues from quality tools like sonarqube and annotate them on your pull requests.
-                 Make sure to have the application.yml located under config/application.yml
-                 """)
+@Command(
+        name = "qualityannotate",
+        mixinStandardHelpOptions = true,
+        subcommands = AutoComplete.GenerateCompletion.class,
+        description =
+                """
+        Helps extracting issues from quality tools like sonarqube and annotate them on your pull requests.
+        Make sure to have the application.yml located under config/application.yml
+        """)
 @RequiredArgsConstructor(onConstructor_ = {@Inject})
 public class MainCommand implements Runnable {
+    private final CodeRepoProvider codeRepoProvider;
+    private final QualityToolProvider qualityToolProvider;
 
-    private final SonarqubeQualityTool sonarqubeApi;
-    private final GithubCodeRepository githubApi;
-    @Parameters(paramLabel = "<qualitytool>", defaultValue = "sonarqube",
-                description = "From which quality-tool you want to retrieve the data. Possible Values: ${COMPLETION-CANDIDATES}; Default: ${DEFAULT-VALUE}",
-                completionCandidates = QualityToolCandidates.class)
+    @Parameters(
+            paramLabel = "<qualitytool>",
+            defaultValue = SonarqubeConfig.NAME,
+            description =
+                    "From which quality-tool you want to retrieve the data. Possible Values: ${COMPLETION-CANDIDATES}; Default: ${DEFAULT-VALUE}",
+            completionCandidates = QualityToolProvider.QualityToolCandidates.class)
     String qualityToolParam;
```

Comment:
⚠️ MAINTAINABILITY: %n should be used in place of \n to produce the platform-specific line separator. [details](https://sonarcloud.io/organizations/quyt/rules?open=java:S3457&rule_key=java:S3457)